lib/replset.js

```diff
@@ -60,8 +80,12 @@ class ReplSet extends EventEmitter {
 
     // Basic config settings for replicaset
     this.version = 1;
+    this.protocolVersion = options.protocolVersion;
     this.replSet = options.replSet;
 
+    // Remove the values from the options
+    delete this.options['protocolVersion'];
```

Comment:
We don't usually mutate objects like this because of how V8 works, also I don't see a reason it would hurt to leave it in the options.

Reply:
@mbroadst The answer to this comment and the 4th comment is that I was just trying to stay consistent. In the constructor, values are taken out from `options` and saved locally. The `options` are then passed as `clientOptions` to each `Server` instance ([here](https://github.com/mongodb-js/mongodb-topology-manager/blob/master/lib/replset.js#L80)). I did a search in the class and there are **no** cases of using values from options besides the constructor.

Here's three spots in the constructor where the values are extracted from options (and in 2/3 of the cases, they are deleted from the options):
* https://github.com/mongodb-js/mongodb-topology-manager/blob/master/lib/replset.js#L28-L29
* https://github.com/mongodb-js/mongodb-topology-manager/blob/master/lib/replset.js#L47-L56
* https://github.com/mongodb-js/mongodb-topology-manager/blob/master/lib/replset.js#L63

My original guess was because the `options` are passed to `Server` and the person wanted to prevent accidental repeat options, but if that doesn't matter to you, then we don't need to do that. Maybe it was to allow the user to easily access these options from the class (i.e. `ReplSet.configSettings` vs `ReplSet.options.configSettings`)

Would you like me to change these occurrences as well (`replSet`, `electionCycleWaitMS`, `configSettings`, & `retryWaitMS`) to stay consistent? (Change them to read directly from options and not store locally)

Also, can you give a brief explanation of what you mean by **because of how V8 works** (interested in learning why)

Reply:
I think the motto for working on these legacy projects is "Don't touch it unless you're already working on the piece of code, or writing new code." So in this case, I would say let's not pull it out and follow old convention, but also let's not fix the old instances unless we're specifically working on those features. 

Regarding "how v8 works", I think this [blog](https://v8.dev/blog/fast-properties) might have the relevant information. Generally, you want to research v8's "hidden classes"